services/worker-manager/src/providers/azure.js

```diff
@@ -148,8 +148,9 @@ class AzureProvider extends Provider {
     // Create "empty" workers to provision in provisionResources loop
     await Promise.all(cfgs.map(async cfg => {
       // This must be unique to currently existing instances and match [a-z]([-a-z0-9]*[a-z0-9])?
+      // 38 chars is workerPoolId / workerId limit
       const poolName = workerPoolId.replace(/[\/_]/g, '-').slice(0, 38);
-      const virtualMachineName = `vm-${poolName}-${nicerId()}-${nicerId()}`.slice(0, 64);
+      const virtualMachineName = `vm-${poolName}-${nicerId()}-${nicerId()}`.slice(0, 38);
```

Comment:
For a 36-char workerPoolId, this would leave one character of entropy for the machine name.  Let's address that in a followup, but ensuring a minimum of entropy and truncating the poolName to accomodate it